outlines/prompts.py

```diff
@@ -34,12 +36,23 @@ def __call__(self, *args, **kwargs) -> str:
         The rendered template as a Python ``str``.
 
         """
-        bound_arguments = self.signature.bind(*args, **kwargs)
-        bound_arguments.apply_defaults()
-        return self.jinja_environment.render(**bound_arguments.arguments)
+        if self.signature is not None:
+            bound_arguments = self.signature.bind(*args, **kwargs)
+            bound_arguments.apply_defaults()
+            return self.template.render(**bound_arguments.arguments)
+        else:
+            return self.template.render(**kwargs)
+
+    @classmethod
+    def from_file(cls, path: Path):
+        """Create a Prompt instance from a file containing a Jinja template.
 
-    def __str__(self):
```

Comment:
I removed the `__str__` implementation from `Prompt` because it now keeps a `jinja2.Template` as its internal state. And, it's not straightforward to serialize a template made from more than one file...

That should be the only breaking change in this PR. If keeping this feature is important, I can work around it by adding a new internal variable to the class to store the`str` version of the template. And when `from_file` is used, it would return the content of the file passed as an argument, regardless of any other files it might depend on... However, I'm still unsure when the `__str__` method would be useful anyway, WDYT?

Reply:
I'm fine with this breaking change